### PR TITLE
feat: interactive app creation

### PR DESCRIPTION
**Pull request by DimitarTachev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [x] Tests for the changes are included.

Related to #3829 

BREAKING CHANGES:
`tns create` without any framework or template flags (--ng, --tsc or --template) will now ask a few questions for selecting the proper template.

### CODE REVIEW

File: lib/commands/create-project.ts [hunk 1]
```diff
@@ -1,44 +1,220 @@
...
+		templates.push({
+			key: CreateProjectCommand.TabsTemplateKey,
+			value: "tns-template-tab-navigation-ts",
+			description: CreateProjectCommand.TabsTemplateDescription
+		});
+		return templates;
+	}
+
+	private getNgFlavors() {
+		const templates: {
```
Comment (Fatme) [inline lib/commands/create-project.ts]:
```
const templates = [
               {
			key: CreateProjectCommand.HelloWorldTemplateKey,
			value: "tns-template-hello-world-ng",
			description: CreateProjectCommand.HelloWorldTemplateDescription
		},
                 {
			key: CreateProjectCommand.DrawerTemplateKey,
			value: "tns-template-drawer-navigation-ng",
			description: CreateProjectCommand.DrawerTemplateDescription
		},
                {
			key: CreateProjectCommand.TabsTemplateKey,
			value: "tns-template-tab-navigation-ng",
			description: CreateProjectCommand.TabsTemplateDescription
		}
];
return templates;
```

File: lib/commands/create-project.ts [hunk 2]
```diff
@@ -1,44 +1,207 @@
...
+			description: CreateProjectCommand.TabsTemplateDescription
+		}];
+
+		return templates;
+	}
+
+	private getNgFlavors() {
+		const templates = [{
+			key: CreateProjectCommand.HelloWorldTemplateKey,
+			value: "tns-template-hello-world-ng",
```
Comment (rosen-vladimirov) [inline lib/commands/create-project.ts]: `RESERVED_TEMPLATE_NAMES.angular`

File: lib/commands/create-project.ts [hunk 3]
```diff
@@ -1,44 +1,207 @@
...
+		if ((!selectedTemplate || !projectName) && isInteractive()) {
+			this.printInteractiveCreationIntro();
+		}
+
+		if (!projectName && isInteractive()) {
+			projectName = await this.$prompter.getString(`${getNextInteractiveAdverb()}, what will be the name of your app?`, { allowEmpty: false });
+			this.$logger.info();
+		}
+
+		projectName = await this.$projectService.validateProjectName({ projectName: projectName, force: this.$options.force, pathToProject: this.$options.path });
```
Comment (rosen-vladimirov) [inline lib/commands/create-project.ts]: The validation must be before `printInteractiveCreationIntro` call, as if it fails the output is strange:
```
tns create ap

# Let’s create a NativeScript app!

Answer the following questions to help us build the right app for you. (Note: you
can skip this prompt next time using the --template option, or the --ng, --vue, --ts,
or --js flags.)

Path already exists and is not empty D:\Rosen\Work\nativescript-cli\scratch\ap
```

File: lib/services/project-service.ts [hunk 4]
```diff
@@ -21,32 +21,44 @@ export class ProjectService implements IProjectService {
...
+		const projectDir = this.getValidProjectDir(opts.pathToProject, projectName);
+		if (this.$fs.exists(projectDir) && !this.$fs.isEmptyDir(projectDir)) {
+			this.$errors.fail("Path already exists and is not empty %s", projectDir);
+		}
+
+		return projectName;
+	}
 
-		const selectedPath = path.resolve(projectOptions.pathToProject || ".");
+	private getValidProjectDir(pathToProject: string, projectName: string): string {
```
Comment (rosen-vladimirov) [inline lib/services/project-service.ts]: private methods should be below public ones

**Pull request merged**